readthedocs/core/middleware.py

```diff
@@ -205,7 +206,7 @@ def process_request(self, request):
                 settings.SESSION_COOKIE_NAME not in request.COOKIES
             ):
                 # Hack request.session otherwise the Authentication middleware complains.
-                request.session = {}
+                request.session = SessionBase()  # create an empty session
```

Comment:
We could instantiate an empty instance of the same class that we are using in the backend.

```
from importlib import import_module
from django.conf import settings
SessionStore = import_module(settings.SESSION_ENGINE).SessionStore
request.session = SessionStore()
```

I suppose that won't be any difference since the `session_key` is `None`, though.